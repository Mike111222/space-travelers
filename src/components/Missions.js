import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import SingleMission from './SingleMission';
import { allMissions, getMissionFromAPI, isLoading } from '../redux/mission/missionSlice';

const Missions = () => {
  const missions = useSelector(allMissions);
  const loading = useSelector(isLoading); // Changed from useDispatch to useSelector

  const dispatch = useDispatch();
  useEffect(() => {
    if (missions.length === 0 && !loading) { // Added check for loading state
      dispatch(getMissionFromAPI());
    }
  }, [dispatch, loading, missions.length]);

  return (
    <section className="missionContainer">
      <table className="missions">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
            <th> </th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <SingleMission key={mission.id} mission={mission} />
          ))}
        </tbody>
      </table>
    </section>
  );
};

export default Missions;
