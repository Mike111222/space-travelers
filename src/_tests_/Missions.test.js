import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Missions from '../components/Missions';

const mockStore = configureStore([]);

describe('Missions', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      missions: {
        missions: [
          {
            id: '1',
            missionName: 'Mission 1',
            description: 'Mission 1 description',
            joined: false,
          },
          {
            id: '2',
            missionName: 'Mission 2',
            description: 'Mission 2 description',
            joined: true,
          },
        ],
        isLoading: false,
      },
    });
  });

  it('should match the snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  // Add more tests as needed to test the rendering of the Missions component and its behavior.
});
