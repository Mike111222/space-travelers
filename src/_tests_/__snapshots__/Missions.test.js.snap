// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions should match the snapshot 1`] = `
<DocumentFragment>
  <section
    class="missionContainer"
  >
    <table
      class="missions"
    >
      <thead>
        <tr>
          <th>
            Mission
          </th>
          <th>
            Description
          </th>
          <th>
            Status
          </th>
          <th>
             
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Mission 1
          </td>
          <td>
            Mission 1 description
          </td>
          <td>
            <div
              class="member-badge"
            >
              NOT A MEMBER
            </div>
          </td>
          <td>
            <button
              class="joinBtn"
              type="button"
            >
              Join Mission
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Mission 2
          </td>
          <td>
            Mission 2 description
          </td>
          <td>
            <div
              class="member-badge active-mission"
            >
              Active Member
            </div>
          </td>
          <td>
            <button
              class="leaveBtn"
              type="button"
            >
              Leave Mission
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</DocumentFragment>
`;
